src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/redundantthis/InputRedundantThisAnonymousObjects.java

```diff
@@ -0,0 +1,62 @@
+package com.puppycrawl.tools.checkstyle.checks.coding.redundantthis;
+
+/*
+* Config = default
+*/
+public class InputRedundantThisAnonymousObjects {
+
+    interface Anon {
+        public void foo();
+    }
+
+    void methodOne() {
+        Anon foo = new Anon() {
+
+            public void foo() {
+                 this.doSideEffect(); // violation
+            }
+
+            public void doSideEffect() {}
+        };
+
```

Comment:
add `this.field = 0;` here. Basically, there should be any use of `this.` __after__ an anonymous class but __before__ the end of the method.

Reply:
Sadly it didn't fix the issue, but changing the signature of `register` method did, thus confirming @strkkk 's intuition. 
Should we keep it this way?

Reply:
This is easy to check. If you comment out the call to the `register` method for an anonymous class in the method `endCollectingDeclarations`, the test `testAnonymousObjects` will fail.
Before this change it succeeded after such change. That is what `RemoveVoidMethod` pitest mutator does.
For now such mutation makes tests to fail, so the suppression is not need any more.

Reply:
Got it. so I guess we could keep it, but **IDEA Inspections Pull Request (Checkstyle)** is not happy with this fix.
![Screenshot from 2020-04-08 12-01-51](https://user-images.githubusercontent.com/24209192/78771709-e99f3c80-7990-11ea-8f0c-27cbb23c60ab.png)

Reply:
Just revert it to `void`

Reply:
Understood, but what about the pitest issue? Do we just declare as an ignored item?

Reply:
> Just revert it to `void`

Done.